CMakeLists.txt

```diff
@@ -789,17 +789,14 @@ endif()
 include(CoverageReport)
 
 # HLSL Change begin - Set default distribution value.
-if (NOT CMAKE_CONFIGURATION_TYPES)
-  set(LLVM_DISTRIBUTION_COMPONENTS "dxc;dxcompiler;dxc-headers" CACHE STRING "")
-endif ()
+set(LLVM_DISTRIBUTION_COMPONENTS "dxc;dxcompiler;dxc-headers" CACHE STRING "")
 # HLSL Change end - Set default distribution value.
 
 # This must be at the end of the LLVM root CMakeLists file because it must run
 # after all targets are created.
 if(LLVM_DISTRIBUTION_COMPONENTS)
-  if(CMAKE_CONFIGURATION_TYPES)
-    message(FATAL_ERROR "LLVM_DISTRIBUTION_COMPONENTS cannot be specified with multi-configuration generators (i.e. Xcode or Visual Studio)")
-  endif()
+  # HLSL Change - Deleted restriction for non-IDE builds. This works with Visual
+  # Studio for DXC because of the limited number of targets.
```

Comment:
Is this because of the number of targets that are in `LLVM_DISTRIBUTION_COMPONENTS` in particular? That is to say, would this work in LLVM in general in cases where people set a very small number of distribution components?

If so, consider changing the check to be related to the size of the list.

Reply:
There is really nothing about the `LLVM_DISTRIBUTION_COMPONENTS` and its targets that can't be configured always. The IDE generators _really_ struggle with large numbers of utility targets, so in upstream (and in DXC) we disable generating a lot of utility targets (like the `install-llc` target) when you're using an IDE generator.

Since DXC's only real supported distribution is three targets, I enabled the per-target install rules for them all the time, and just made them always grouped as a distribution.